File: solutions/0169-majority-element/solution.java
import java.util.Arrays;
class Solution {
    public int majorityElement(int[] arr) {
        Arrays.sort(arr);
        return arr[arr.length/2];
//         int n=arr.length;
//         for(int i=0;i<n;i++){
//             if(arr[i]>arr[i+1]){
//                 int temp=arr[i];
//                 arr[i]=arr[i+1];
//                 arr[i+1]=temp;
//             }
//         }
//         int ans=arr[0]; int freq=1;
//         for(int i=0;i<n;i++){
//             if(arr[i]==arr[i-1]) freq++;
//             else{
//                 freq=1;
//                 ans=arr[i];
//             }
//         }
//         if(freq>n/2) return ans;
    }
    
}

File: solutions/0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int count=1;
        int ele=nums[0];
        for(int i=1;i<n;i++){
            if(ele!=nums[i] && count==1){
                ele=nums[i];
            }
            else if(ele!=nums[i]) count--;
            else count++;
        }
        return ele;
    }
};
